tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1223,26 +1256,52 @@ class CudnnRnnSequenceTensorDescriptor
         /*tensorDesc=*/tensor_desc.get(), /*dataType=*/data_type,
         /*nbDims=*/sizeof(dims) / sizeof(dims[0]), /*dimA=*/dims,
         /*strideA=*/strides));
+#if CUDNN_VERSION >= 7201
+    if (seq_lens == nullptr) {
+      return CudnnRnnSequenceTensorDescriptor(parent, seq_length, batch_size,
+                                              data_size, data_type,
+                                              std::move(tensor_desc));
+    } else {
+      RNNDataDescriptor data_desc = CreateRNNDataDescriptor();
+      float padding_fill = 0.0f;
+      RETURN_IF_CUDNN_ERROR(cudnnSetRNNDataDescriptor(
+          /*RNNDataDesc=*/data_desc.get(), /*dataType*/ data_type,
+          /*layout=*/CUDNN_RNN_DATA_LAYOUT_SEQ_MAJOR_UNPACKED,
+          /*maxSeqLength=*/seq_length,
+          /*batchSize=*/batch_size, /*vectorSize=*/data_size,
+          /*seqLengthArray=*/seq_lens,
+          /*paddingFill*/ (void*)&padding_fill));
+      return CudnnRnnSequenceTensorDescriptor(parent, seq_length, batch_size,
+                                              data_size, seq_lens, data_type,
+                                              std::move(data_desc),
+                                              std::move(tensor_desc));
```

Comment:
I guess `tensor_desc` is not needed for var seq length version?

Reply:
`tensor_desc` is still needed to get the params size (i.e., cudnnGetRNNParamsSize). There is no API yet to accept the cudnnRNNDataDescriptor_t `data_desc` to get the params size.